### ISSUE TITLE
Docker travis job failing

### ISSUE THREAD

**Issue by jayvdb** [jantman/awslimitchecker#453]
note "/usr/local/lib/python3.8" in the following log
https://travis-ci.org/jantman/awslimitchecker/jobs/622663576
```py
$ docker run -it --rm -e AWS_DEFAULT_REGION=us-east-1 jantman/awslimitchecker:test_b23dd86_1575899074 --version

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 791, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (python-dateutil 2.8.1 (/usr/local/lib/python3.8/site-packages), Requirement.parse('python-dateutil<2.8.1,>=2.1; python_version >= "2.7"'), {'botocore'})
```

**Comment by jayvdb** [comment]
Also note I am seeing Python 3.7 failures at https://build.opensuse.org/package/show/home:jayvdb:py-new/python-awslimitchecker , which look to be related to pytest versions.

**Comment by jantman** [comment]
@jayvdb Apologies for the delay, and thank you for reporting this.

Yes, it appears that the TravisCI docker build job is failing. It appears that a new version of botocore has been released, which declares a `python-dateutil` version requirement incompatible with the version that awslimitchecker itself uses. I'll work on a fix for this sometime in the next few days.

Regarding the python versions, this is working as intended. The TravisCI VM is running Python 3.7, which then triggers a build of a Docker image based on the `python:3.8-alpine` base image. The 3.7 version is just used to run `tox`, which is controlling the Docker build.

Regarding the openSuSE build failures, I have to admit I'm not terribly familiar with openSuSE these days... I haven't used the distro since version 11.3, which appears to have been just shy of a decade ago. But I do think that I've figured out the issues:

1. The TravisCI builds of master are working, when using pytest 5.3.2. The openSuSE builds you linked are using pytest 4.6.6. awslimitchecker probably needs a test dependency on pytest `>= 5.0.0` or similar. Please note that all of the test-time dependencies specified [in tox.ini](https://github.com/jantman/awslimitchecker/blob/master/tox.ini#L4-L22) are assumed to be installed at the absolute latest version if not specified otherwise.
2. In the [spec file](https://build.opensuse.org/package/view_file/home:jayvdb:py-new/python-awslimitchecker/python-awslimitchecker.spec?expand=1) I see the `%check` phase running the `%pytest` macro without any arguments. Going by the [documentation of what that macro does](https://github.com/openSUSE/python-rpm-macros), i.e. just calling ``pytest``, that is almost certainly a problem, and will absolutely definitely fail, always, period. Awslimitchecker's [testing](https://awslimitchecker.readthedocs.io/en/latest/development.html#unit-testing) uses the [tox](https://tox.readthedocs.org/) test runner, and the pytest suite is only designed to be run via tox. If running with tox isn't an option, you'll need to ensure that pytest is called with the same arguments as how tox would call it. At a minimum, for an OS package, only the test suite for the python version you're building against should be run.

awslimitchecker currently has 13 tox environments, most of which are meaningless for an OS package:

* ``py27,py34,py35,py36,py37,py38,pypy,pypy3`` (version-specific unit test suites) - OS packages should only run one of these, for the Python version you're packaging against.
* docs - only relevant for upstream releases, when building docs for readthedocs.org
* localdocs - only relevant for developers
* integration,integration3 - only runnable on TravisCI, period. Will fail everywhere else in the world.
* docker - only relevant when building the Docker image for hub.docker.com

Side note: In the [spec file](https://build.opensuse.org/package/view_file/home:jayvdb:py-new/python-awslimitchecker/python-awslimitchecker.spec?expand=1) I see the License line identified as "unknown". awslimitchecker is licensed under the GNU Affero General Public License (AGPL) version 3.0 or later, which has a SPDX identifier of ``AGPL-3.0-or-later``. It is of paramount importance that this license be identified correctly.

**Comment by jayvdb** [comment]
I've updated the license. thx. I'll re-attempt the packaging this weekend - im pretty sure it is a pytest version compatibility problem.

**Comment by jantman** [comment]
This has been released in [8.0.1](https://github.com/jantman/awslimitchecker/releases/tag/8.0.1) and is now [live on PyPI](https://pypi.org/project/awslimitchecker/). Thank you!

**Comment by jayvdb** [comment]
Thanks.  Package updated.  I also found the solution to the pytest problems, but now I have all tests failing due to accessing the internet.  I will keep plugging away at it.
